core/src/main/java/org/elasticsearch/transport/ConnectionProfile.java

```diff
@@ -59,6 +63,17 @@ private ConnectionProfile(List<ConnectionTypeHandle> handles, int numConnections
         private final List<ConnectionTypeHandle> handles = new ArrayList<>();
         private final Set<TransportRequestOptions.Type> addedTypes = EnumSet.noneOf(TransportRequestOptions.Type.class);
         private int offset = 0;
+        private TimeValue connectTimeout;
+
+        /**
+         * Sets a connect timeout for this connection profile
+         */
+        public void setConnectTimeout(TimeValue timeout) {
+            if (timeout.millis() < 0) {
+                throw new IllegalArgumentException("timeout must be positive but was: " + timeout);
```

Comment:
The condition (less than zero) and the message (positive) are at odds with each other. Either the condition should be less than or equal to zero, or the message should say non-negative.